akka-typed/src/main/scala/akka/typed/Behavior.scala

```diff
@@ -80,6 +80,77 @@ abstract class ExtensibleBehavior[T] extends Behavior[T] {
 object Behavior {
 
   /**
+   * Return this behavior from message processing in order to advise the
+   * system to reuse the previous behavior. This is provided in order to
+   * avoid the allocation overhead of recreating the current behavior where
+   * that is not necessary.
+   */
+  def Same[T]: Behavior[T] = SameBehavior.asInstanceOf[Behavior[T]]
+  /**
+   * Java DSL: Return this behavior from message processing in order to advise the
+   * system to reuse the previous behavior. This is provided in order to
+   * avoid the allocation overhead of recreating the current behavior where
+   * that is not necessary.
+   */
+  def same[T]: Behavior[T] = Same
```

Comment:
I see no point in having both here. I opt for `same`. In `Actor` it makes sense to have it with capital letter for Scala since the other types are like that. `Immutable`, `Mutable`, `Deferred`.